### PR TITLE
feat: add get_by_id to retrieve row_ids for deletion of documents in...

### PR DESCRIPTION
**Pull request by Daniel199438**
…supabase vector stores and complete delete function for supabase vector stores

# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)
6735

## Type of Change
added functionality

Please delete options that are not relevant.

- [x] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)

# How Has This Been Tested?

Please describe the tests that you ran to verify your changes. Provide instructions so we can reproduce. Please also list any relevant details for your test configuration

- [x] I stared at the code and made sure it makes sense

# Suggested Checklist:

- [x] I have performed a self-review of my own code
- [x] I have commented my code, particularly in hard-to-understand areas

### CODE REVIEW

File: llama_index/vector_stores/supabase.py [hunk 1]
```diff
@@ -107,19 +107,45 @@ def add(self, embedding_results: List[NodeWithEmbedding]) -> List[str]:
...
+            :param ref_doc_id (str): document id
 
         """
-        raise NotImplementedError("Delete not yet implemented for vecs.")
+        row_ids = self.get_by_id(ref_doc_id)
+
+        if len(row_ids) > 0:
+            self._collection.delete(row_ids)
+
+        # raise NotImplementedError("Delete not yet implemented for vecs.")
```
Comment (logan-markewich) [inline llama_index/vector_stores/supabase.py:143]: nit: remove commented code

File: llama_index/vector_stores/types.py [hunk 2]
```diff
@@ -158,6 +158,10 @@ def add(
         """Add embedding results to vector store."""
         ...
 
+    def get_by_id(self, doc_id: str) -> List[str]:
```
Comment (logan-markewich) [inline llama_index/vector_stores/types.py:161]: I don't think we want to add this to the base class just yet -- I think there will be many errors about this not being implemented by the other classes

**Comment by logan-markewich** [comment]
A few linting/unit tests to clean up, but makes sense otherwise!

**Comment by logan-markewich** [comment]
@Daniel199438 are you still interested in fixing up the PR?

**Comment by Daniel199438** [comment]
@logan-markewich Yes of course, Im sorry I was the last weeks in vacation and had elsewhere other to-do's. 
I will fix the issues until the end of this week!

**Comment by logan-markewich** [comment]
Thanks @Daniel199438 !

**Pull request merged**